internal/service/streamprocessor/model_test.go

```diff
@@ -127,34 +142,76 @@ func optionsToTFModel(t *testing.T, options *admin.StreamsOptions) types.Object
 }
 
 func TestDSSDKToTFModel(t *testing.T) {
-	testCases := []struct {
+	testCases := map[string]struct {
 		sdkModel        *admin.StreamsProcessorWithStats
 		expectedTFModel *streamprocessor.TFStreamProcessorDSModel
-		name            string
 	}{
-		{
-			name: "afterCreate",
+		"afterCreate": {
 			sdkModel: admin.NewStreamsProcessorWithStats(
 				processorID, processorName, []any{pipelineStageSourceSample, pipelineStageEmitLog}, stateCreated,
 			),
 			expectedTFModel: streamProcessorDSTFModel(t, stateCreated, "", optionsToTFModel(t, nil)),
 		},
-		{
-			name:            "afterStarted",
+		"afterStarted": {
 			sdkModel:        streamProcessorWithStats(t, nil),
 			expectedTFModel: streamProcessorDSTFModel(t, stateStarted, statsExample, optionsToTFModel(t, nil)),
 		},
-		{
-			name:            "withOptions",
+		"withOptions": {
 			sdkModel:        streamProcessorWithStats(t, &streamOptionsExample),
 			expectedTFModel: streamProcessorDSTFModel(t, stateStarted, statsExample, optionsToTFModel(t, &streamOptionsExample)),
 		},
 	}
 
-	for _, tc := range testCases {
-		t.Run(tc.name, func(t *testing.T) {
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
 			sdkModel := tc.sdkModel
-			resultModel, diags := streamprocessor.NewTFStreamprocessorDSModel(t.Context(), projectID, instanceName, sdkModel)
+			resultModel, diags := streamprocessor.NewTFStreamprocessorDSModel(t.Context(), projectID, "", workspaceName, sdkModel)
+			if diags.HasError() {
+				t.Fatalf("unexpected errors found: %s", diags.Errors()[0].Summary())
+			}
```

Comment:
```suggestion
			assert.False(t, diags.HasError())
```

in general you shoudn't need to use `Fatal`